### PR TITLE
feat: Add support for env vars in templates

### PR DESCRIPTION
**Pull request by rasa**
### What does this PR do?

Allows `{{ env.HOME }}` in templates.

### What issues does this PR fix or reference?

This allows us to replace
```
{% if yadm.os == "Darwin" %}
home="/Users/{{ yadm.user }}"
{% else %}
home="/home/{{ yadm.user }}"
{% endif %}
```
with
```
home="{{ env.HOME }}"
```

### Previous Behavior

See above.

### New Behavior

See above.

### Have [tests][1] been written for this change?

No. Sorry, I'm not skilled in writing tests in python.

### Have these commits been [signed with GnuPG][2]?

Yes.

---

[1]: https://github.com/TheLocehiliosan/yadm/blob/master/.github/CONTRIBUTING.md#test-conventions
[2]: https://help.github.com/en/articles/signing-commits
[3]: https://github.com/TheLocehiliosan/yadm/blob/master/.github/CONTRIBUTING.md

### CODE REVIEW

**Comment by pkoch** [comment]
@rasa: See https://github.com/rasa/yadm/pull/2 for some tests.

@TheLocehiliosan: Would that be enough to accept the PR?

File: test/test_unit_template_default.py [hunk 1]
```diff
@@ -202,3 +202,22 @@ def test_include(runner, yadm, tmpdir):
...
+
+    script = f"""
+        YADM_TEST=1 source {yadm}
+        set_awk
+        template_default "{input_file}" "{output_file}"
+    """
+    run = runner(command=['bash'], inp=script)
+    assert run.success
+    assert run.err == ''
+    assert output_file.read() == os.environ['PWD']
```
Comment (pkoch) [inline test/test_unit_template_default.py:223]:
```suggestion
    assert output_file.read().strip() == os.environ['PWD']
```

**Review by pkoch — COMMENTED** [review]
Derp, I thought I saw this passing. Sorry, folks. Here's how to fix it.

**Comment by TheLocehiliosan** [comment]
@pkoch - thanks for the tests. I'll try to include this PR before the release of 3.2.0.

**Comment by TheLocehiliosan** [comment]
I signed these commits, and merged this into develop.

**Pull request closed without merging**